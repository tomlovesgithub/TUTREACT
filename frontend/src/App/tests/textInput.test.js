import React from 'react'
import TextInput from '../components/TextInput.js'

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow } from 'enzyme'

Enzyme.configure({ adapter: new Adapter()})

describe('textInput', () => {

  it('renders without crashing', () => {
    const component = mount(<TextInput/>);
    expect(component).toMatchSnapshot();
  });

  it('value passed as props', () => {
    const component = shallow(<TextInput value='hello'/>);
    expect(component.find('textarea').props().value).toEqual('hello');
  });

  it('no value passed as props', () => {
    const component = shallow(<TextInput/>);
    expect(component.find('textarea').props().value).toEqual('');
  });

  it('ID passed as props', () => {
    const component = shallow(<TextInput id='testID'/>);
    expect(component.find('textarea').props().id).toEqual('testID');
  });

  it('no ID passed as props', () => {
    const component = shallow(<TextInput/>);
    expect(component.find('textarea').props().id).toEqual(undefined);
  });

  it('value stored in state', () => {
    const component = shallow(<TextInput id='testID'/>);
    expect(component.find('textarea#testID').props().value).toEqual('');
    expect(component.state('value')).toEqual('');
    component.find('textarea#testID').simulate('change', { target: { value: 'HelloWorld' } })
    expect(component.find('textarea#testID').props().value).toEqual('HelloWorld');
    expect(component.state('value')).toEqual('HelloWorld');
  });

});
