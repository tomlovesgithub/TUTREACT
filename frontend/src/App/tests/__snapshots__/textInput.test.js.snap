// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textInput renders without crashing 1`] = `ReactWrapper {}`;
